src/components/PopupWithForm.js

```diff
@@ -7,6 +7,11 @@ class PopupWithForm extends Popup {
     this._handleFormSubmit = handleFormSubmit;
   }
 
+  open() {
+    super.open();
+    this.revertSaveButtonListener();
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.